### DISCUSSION TITLE
anyone, pls share "backup strategy" (incl thoughts and tools) for your digital memories

### DISCUSSION THREAD

**Discussion by onetimecontributor** [io7m-com/exfilac#30] [Q&A]
Hi, I'm in the process of deciding on how to replace my existing special-purpose online storage for digital memories.
I want to get rid of privacy-concirning service with all all those "your highlights" etc, silent tagging for "convenient search" purpose etc., but trying to keep certain level of convenience in using it day-to-day and keeping it cost-efficient (for example google's 2TB plan is $100/yr, I currently have less than 1TB).

I will appreciate if community shares their experience and approach.

What I came up so far and trying to estimate costs and find respective FOSS tools for Android:
0. "final" data prep/cleanup to happen on a single device where number of clients will push their data.
1. Online access from different devices (encrypted) - using smth like S3 Glacier Instant Retrieval in one geographical region (closer to clients)
2. Online primary backup (encrypted) - using smth like S3 Glacier Deep Archive with copying data from "online" instance, should use distant geographical region other than for "online".
3. Offline backup (encrypted) - blu-ray disks 25GB or 50GB, no redundancy

Tools I need to investigate to implement those "stages":
0. Syncthing (for data sync, including encryption); no clean-up Android app found so far (only self-hosted solutions like photoprism and immich), so manual clean up atm
1. exfilac
2. AWS management console
3. have no idea atm if that's doable with Android. so far it doesn't look so.

Previous thoughts that I've rejected (thought of avoiding cloud as much as possible):
1. Keep the backup only offline on blu-ray m-drives, with redundancy - 2 copies, record/keep second copy at another geographical location.
Reason to abandon: logistics (remote operations, etc); cost efficiency; risks of loosing smth while waiting for content to fill another 25GB/50GB disk, per-GB cost may not be heaven lower if to account for extra recording device and disks.
2. Keep the "online" version on-premises by storing it on a HDD/SSD drive attached to online device.
Reason to abandon: privacy risks, as I'll need to take care of security; not as cost efficient as current solution.

@io7m , thanks for your solution that may make my plan achievable!

**Comment by io7m** [comment]
I currently run `exfilac` on a couple of devices, and both of those devices are configured with a [wireguard](https://www.wireguard.com/) connection to a physical server that I control. That physical server is running [MinIO](https://min.io/), and that physical server is backed up nightly to another physical server with a large ZFS array.

I'm considering also uploading encrypted copies of the files in `minio` to [Vultr object storage](https://docs.vultr.com/vultr-object-storage), but I haven't gotten around to this yet.

**Reply by onetimecontributor** [reply]
> I'm considering also uploading encrypted copies of the files in `minio` to [Vultr object storage](https://docs.vultr.com/vultr-object-storage), but I haven't gotten around to this yet.

do you by chance know if vultr is more cost-efficient than aws s3 "glaciar deep archive" storage class (it seems it's heavily optimized towards such need)?

**Reply by io7m** [reply]
I think it probably isn't competitive on raw price, but it would definitely be competitive on price _transparency_. Vultr consistently use fixed, transparent pricing, so there's no risk of "Oops, I accidentally ran up a $100000 AWS bill". :slightly_smiling_face: 

https://youtu.be/ynueyo_Efjo
